File: web/React/get-start/src/qflow/component/QflowBpmn/lib/customDescriptors/qflow.js
const qflowModdleDescriptor = {

    "name": "qflow",
    "uri": "http://qflow",
    "prefix": "qflow",
    "xml": {
        "tagAlias": "lowerCase"
    },
    "types": [
        {
            "name": "InputOutputParameterDefinition",
            "isAbstract": true
        },
        {
            "name": "FormData",
            "superClass": [ "Element" ],
            "meta": {
              "allowedIn": [
                "bpmn:FlowNode"
              ]
            },
            "properties": [
              {
                "name": "fields",
                "type": "FormField",
                "isMany": true
              },
              {
                "name": "businessKey",
                "type": "String",
                "isAttr": true
              }
            ]
        }, 
        {
            "name": "FormField",
            "superClass": [ "Element" ],
            "properties": [
              {"name": "id","type": "String","isAttr": true},
              {"name": "name","type": "String","isAttr": true,"default": "未命名" },
              {"name": "link", "type": "String", "isAttr": true,"default" : ""},
              {"name": "submit_var_name", "type": "String", "isAttr": true,"default" : ""},
              {"name": "description", "type": "String", "isAttr": true,"default" : ""},
              {"name": "field_type","type": "String","isAttr": true,"default": "input"},
              {"name": "default_value", "type": "String", "isAttr": true, "default" : ""},
              {"name": "values","type": "Value","isMany": true},
              {"name": "is_required", "type": "Boolean", "isAttr": true, "default" : true},
              {"name": "is_on_task_top", "type": "Boolean", "isAttr": true, "default" : false},
              {"name": "is_on_apply_fill", "type": "Boolean", "isAttr": true, "default" : false},
            ]
        }, 
        {
            "name": "Value",
            "superClass": [
              "InputOutputParameterDefinition"
            ],
            "properties": [
              {
                "name": "name",
                "isAttr": true,
                "type": "String"
              },
              {
                "name": "value",
                "isBody": true,
                "type": "String"
              }
            ]
        },    
        {
          "name": "BewitchedFlowNode",
          "extends": [
            "bpmn:FlowNode"
          ],
          "properties": [
            {
                "name": "description",
                "type": "String", 
                "isAttr": true, 
                "default" : ""
            },
            {
                "name": "link",
                "type": "String", 
                "isAttr": true, 
                "default" : ""
            },
            {
                "name": "submit_var_name",
                "type": "String", 
                "isAttr": true, 
                "default" : ""
            },
            {
                "name": "owner",
                "type": "String", 
                "isAttr": true, 
                "default" : ""
            },
            {
                "name": "is_skip",
                "type": "String", 
                "isAttr": true, 
                "default" : "N"
            }
          ]
        },
        {
          "name": "ManualTask",
          "superClass": [ "Element" ],
          "meta": {
            "allowedIn": [
              "bpmn:ManualTask"
            ]
          },
          "properties": [
            {
              "name": "name",
              "type": "String",
              "isAttr": true,
              "default" : "审批"
            },
            {
              "name": "success",
              "type": "String",
              "isAttr": true,
              "default": "通过"
            },
            {
              "name": "fail",
              "type": "String",
              "isAttr": true,
              "default" : "不通过"
            }
          ]
      },
      {
        "name": "ServiceTask",
        "superClass": [ "Element" ],
        "meta": {
          "allowedIn": [
            "bpmn:ServiceTask"
          ]
        },
        "properties": [
          {
            "name": "tool_id",
            "type": "String",
            "isAttr": true,
            "default" : ""
          }
        ]
      },
      {
        "name": "ParallelGateway",
        "superClass": [ "Element" ],
        "meta": {
          "allowedIn": [
            "bpmn:ParallelGateway"
          ]
        },
        "properties": [
          {
            "name": "name",
            "type": "String",
            "isAttr": true,
            "default" : "分支"
          },
          { "name": "is_on_apply_fill", "type": "Boolean", "isAttr": true, "default" : false},
          { "name": "is_required", "type": "Boolean", "isAttr": true, "default" : true}
        ]
      }
  ]

}

export default qflowModdleDescriptor;

File: Interpreter/Python/Xml/Bpmn/python_camunda/descriptor/qflow.py
#coding:utf-8

qflow_descriptor = {

    "shape" : [
        
        {"type" : "bpmn2:startEvent", "category" : "start"},

        {"type" : "bpmn2:userTask", "category" : "task"}, 
        
        {"type" : "bpmn2:parallelGateway", "category" : "gateway"}, 
        
        {"type" : "bpmn2:serviceTask", "category" : "task"}, 
        
        {"type" : "bpmn2:manualTask", "category" : "task"}
        
    ], 

    "connection" : { "type" : "bpmn2:sequenceFlow" },

    "content" : { 
        
        "type" : "qflow:formField",

        "attrs" : [

            {"name" : "id"},
            {"name" : "name", "default" : u"未命名"},
            {"name" : "link", "default" : u""},
            {"name" : "submit_var_name", "default" : u""},
            {"name" : "description", "default" : u""},
            {"name" : "field_type", "default" : u"input"},
            {"name" : "default_value", "default" : u""},
            {"name" : "is_required", "default" : u"Y"},
            {"name" : "is_on_task_top", "default" : u"N"},
            {"name" : "is_on_apply_fill", "default" : u"N"},
            {"name" : "is_pushed", "default" : u"N"},
            {"name" : "is_inherited", "default" : u"N"},
            {"name" : "is_task_name", "default" : u"N"},
            {"name" : "is_encrypt", "default" : u"N"},
            
        ]
        
    },

    "attrs" : [ 
        
        {"name" : "name", "default" : u"未命名"},
        {"name" : "description", "default" : u""},
        {"name" : "link" , "default" : u""},
        {"name" : "group_name" , "default" : u""},
        {"name" : "submit_var_name" , "default" : u""},
        {"name" : "owner" , "default" : u""},
        {"name" : "is_task_owner" , "default" : "0"},
        {"name" : "is_skip" , "default" : "N"}
        
    ],

    "extension" : {

        "bpmn2:serviceTask" : {
            
            "type" : "qflow:serviceTask",

            "attrs" : [

                {"name" : "tool_id", "default" : u"-1"},
                
            ]

        },

        "bpmn2:manualTask" : {
            
            "type" : "qflow:manualTask",

            "attrs" : [


                {"name" : "success", "default" : u"通过"},
                {"name" : "fail", "default" : u"不通过"},
                
            ]

        }
            
    }

    
    #"annotation" : {
    #    "type" : "bpmn2:textAnnotation",
    #    "connection" : {"type" : "bpmn2:association"},
    #    "content" : {"type" : "bpmn2:text"}
    #}
        
        
}
